grakn-graql/src/main/java/ai/grakn/graql/internal/query/QueryOperationExecutor.java

```diff
@@ -424,35 +428,35 @@ GraknTx tx() {
      */
     @AutoValue
     static abstract class VarAndProperty {
+
         abstract Var var();
         abstract VarPropertyInternal property();
 
         static VarAndProperty of(Var var, VarProperty property) {
-            return new AutoValue_InsertQueryExecutor_VarAndProperty(var, VarPropertyInternal.from(property));
+            return new AutoValue_QueryOperationExecutor_VarAndProperty(var, VarPropertyInternal.from(property));
         }
 
         static Stream<VarAndProperty> fromPattern(VarPatternAdmin pattern) {
             return pattern.getProperties().map(prop -> VarAndProperty.of(pattern.var(), prop));
         }
 
-        void insert(InsertQueryExecutor executor) {
-            property().insert(var(), executor);
-        }
-
-        void define(InsertQueryExecutor executor) {
-            property().define(var(), executor);
-        }
-
-        Set<Var> requiredVars() {
-            return property().requiredVars(var());
-        }
-
-        Set<Var> producedVars() {
-            return property().producedVars(var());
+        private PropertyExecutor executor(ExecutionType executionType) {
+            switch (executionType) {
```

Comment:
Agreed. I've now learned you can add abstract methods to enums so I'll do that 👍